### PR TITLE
Spread replicas with custom resources in torch tune serve release test

### PR DESCRIPTION
**Pull request by zcin**
Spread replicas with custom resources in torch tune serve release test

For the Golden Notebook Torch Tune Serve release test.
Use custom resources to make sure one replica gets scheduled for each of the two nodes in the cluster.

Signed-off-by: Cindy Zhang <cindyzyx9@gmail.com>

### CODE REVIEW

File: release/golden_notebook_tests/workloads/torch_tune_serve_test.py [hunk 1]
```diff
@@ -180,7 +180,10 @@ def setup_serve(model, use_gpu: bool = False):
     )  # Start on every node so `predict` can hit localhost.
     serve.run(
         MnistDeployment.options(
-            num_replicas=2, ray_actor_options={"num_gpus": bool(use_gpu)}
+            num_replicas=2,
+            ray_actor_options={"num_gpus": 1, "resources": {"worker": 1}}
+            if use_gpu
+            else {},
```
Comment (edoakes) [inline release/golden_notebook_tests/workloads/torch_tune_serve_test.py:186]: this won't spread across nodes if no GPU is used, is that intentional?

Comment (zcin) [reply release/golden_notebook_tests/workloads/torch_tune_serve_test.py:186]: Yeah I think `no_gpu` is used for smoke test. There is only one worker node type in this test, so no gpu should mean all replicas get started on head node.

Comment (edoakes) [reply release/golden_notebook_tests/workloads/torch_tune_serve_test.py:186]: ok

**Review by edoakes — APPROVED** [review]

**Review by aslonnie — APPROVED** [review]
(if ed says it is good, it is good)

<div><br/>via <a href="https://giphy.com/gifs/shaun-the-sheep-movie-not-my-gif-2016-oscar-nominations-tIeCLkB8geYtW">GIPHY</a></div>

**Comment by zcin** [comment]
@edoakes ready to merge?

**Pull request merged**